gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/MultiLicenseHeaderStep.java

```diff
@@ -55,113 +60,95 @@
-            }
-            this.licenseHeader = licenseHeader1;
-            this.hasYearToken = licenseHeader1.contains("$YEAR");
-            if (this.hasYearToken) {
-                int yearTokenIndex = licenseHeader1.indexOf("$YEAR");
-                this.licenseHeaderBeforeYearToken = licenseHeader1.substring(0, yearTokenIndex);
-                this.licenseHeaderAfterYearToken = licenseHeader1.substring(yearTokenIndex + 5);
-                this.licenseHeaderWithYearTokenReplaced = licenseHeader1.replace(
-                        "$YEAR", String.valueOf(YearMonth.now().getYear()));
-                this.yearMatcherPattern = Pattern.compile("[0-9]{4}(" + Pattern.quote(yearSeparator) + "[0-9]{4})?");
-            }
+        static LicenseHeader fromTemplate(String template) {
+            String unixEndings = LineEnding.toUnix(template.trim());
+            Iterable<String> lines = Splitter.on('\n').trimResults().split(unixEndings);
+            String javadocHeader = Streams.stream(lines)
+                    .map(line -> line.isEmpty() ? " *" : " * " + line)
+                    .collect(Collectors.joining("\n"));
+            return new LicenseHeader("/*\n" + javadocHeader + "\n */\n");
         }
 
-        private boolean matchesLicenseWithYearToken(String existingHeader) {
+        private boolean matches(String existingHeader) {
             int startOfTheSecondPart = existingHeader.indexOf(licenseHeaderAfterYearToken);
             return startOfTheSecondPart > licenseHeaderBeforeYearToken.length()
                     && (existingHeader.startsWith(licenseHeaderBeforeYearToken)
                             && startOfTheSecondPart + licenseHeaderAfterYearToken.length() == existingHeader.length())
-                    && yearMatcherPattern
+                    && YEAR_RANGE
                             .matcher(existingHeader.substring(
                                     licenseHeaderBeforeYearToken.length(), startOfTheSecondPart))
                             .matches();
         }
 
-        private boolean matches(String existingHeader) {
-            if (hasYearToken) {
-                // that means we have the license like `licenseHeaderBeforeYearToken 1990-2015
-                // licenseHeaderAfterYearToken`
-                return matchesLicenseWithYearToken(existingHeader);
-            }
-            return existingHeader.equals(licenseHeader);
-        }
-
-        private String render() {
-            if (hasYearToken) {
-                return licenseHeaderWithYearTokenReplaced;
-            }
-            return licenseHeader;
+        private String render(String existingHeader) {
+            Matcher yearInfo = YEAR_RANGE.matcher(existingHeader);
+            boolean existingHeaderContainsYear = yearInfo.find();
+            String year = existingHeaderContainsYear
+                    ? yearInfo.group(0)
+                    : String.valueOf(YearMonth.now().getYear());
```

Comment:
this line is really the money. 

the rest of the PR is mainly just me deleting indirection that doesn't seem necessary now that we're not working around spotless' API